File: 05_hash_tables/ruby/02_check_voter.rb
# in order to be accessible from within the check_voter function
# voted has to be an instance variable (or global / class)
@voted = {}

def check_voter(name)
  if @voted.key?(name) && @voted[name]
    puts "kick them out!"
  else
    @voted[name] = true
    puts "let them vote!"
  end
end

check_voter("tom")
check_voter("mike")
check_voter("mike")

File: 05_hash_tables/php/02_check_voter.php
<?php

$voted = [];

function check_voter($name)
{
    global $voted;
    if (isset($voted[$name])) {
        print "kick them out!\n";
    } else {
        $voted[$name] = true;
        print "let them vote!\n";
    }
}

check_voter("tom"); // let them vote!
check_voter("mike"); // let them vote!
check_voter("mike"); // kick them out!
